.github/workflows/build-sample-apps.yml

```diff
@@ -67,13 +67,37 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
+    - name: Set Default Firebase Distribution Groups
+      run: |
+        # Define all the possible distribution groups
+        ALL_BUILDS_GROUP="all-builds"
+        FEATURE_BUILDS_GROUP="feature-branch"
+        STABLE_BUILDS_GROUP="next"
+
+        # Initialize with the default distribution group
+        distribution_groups=("$ALL_BUILDS_GROUP")
+
+        # Determine current app type and Git context
+        is_primary_app=$([[ "${{ matrix.apn-or-fcm }}" == "APN" ]] && echo "true" || echo "false")
+        current_branch="${GITHUB_REF}"
+
+        # Append distribution groups based on branch and context if the app is primary
+        if [[ "$is_primary_app" == "true" ]]; then
+          [[ "$current_branch" == "refs/heads/feature/"* ]] && distribution_groups+=("$FEATURE_BUILDS_GROUP")
+          [[ "$current_branch" == "refs/heads/main" ]] && distribution_groups+=("$STABLE_BUILDS_GROUP")
+        fi
+
+        # Export the groups as an environment variable
+        echo "firebase_distribution_groups=$(IFS=','; echo "${distribution_groups[*]}")" >> $GITHUB_ENV
```

Comment:
I tested [final value with dummy input](https://github.com/customerio/customerio-ios/actions/runs/12693036897/job/35379733422#step:5:369), and it seems to be working well.